packages/internal/src/types-internal.d.ts

```diff
@@ -0,0 +1,5 @@
+/**
+ * For use by debug.js but defined here because .js files cannot define
+ * circular TS types.
+ */
+export type OptPrefixPath = undefined | [string, OptPrefixPath];
```

Comment:
Flip this even though it is a less familiar way to visualize linked lists. The remainder of the list is "earlier", upward in the creation tree, and leftward in the rendered prefix path.

Reply:
> O(1) for each sub-tracer creation.

Given that tracer prefixes are going to be shorter than half a line (otherwise the logs become unreadable), I doubt it's worth the complexity. Reviewing this is going to take a while.

Reply:
@dckc , at #11857 , your suggestion here turned out to be an AWESOME improvement. I'm embarrassed I resisted at first. THANKS! Resolving.